pkg/planner/core/binary_plan_test.go

```diff
@@ -124,6 +124,10 @@ func TestTooLongBinaryPlan(t *testing.T) {
 	require.NoError(t, logutil.InitLogger(newCfg.Log.ToLogConfig()))
 	store := testkit.CreateMockStore(t)
 	tk := testkit.NewTestKit(t, store)
+	if tk.MustQuery("select @@tidb_schema_cache_size > 0").Equal(testkit.Rows("1")) {
+		t.Skip("TODO: the performance is poor for this test under infoschema v2")
```

Comment:
https://github.com/pingcap/tidb/pull/52479 optimize the performance here, after that we do not need to skip.